pages/task/read/views/project.jsx

```diff
@@ -17,45 +22,112 @@ const Project = ({ task, project, establishment, children, schema, formFields })
-        <p><Snippet type={task.type}>versions.submitted.hint</Snippet></p>
-        <p>
-          <Link
-            page="project.version.read"
-            className="govuk-button button-secondary"
-            versionId={submitted}
-            establishmentId={establishment.id}
-            projectId={project.id}
-            label={<Snippet>versions.submitted.label</Snippet>}
-          />
-        </p>
-
-        {
-          task.status === 'with-inspectorate' &&
-            <Fragment>
-              <p><Snippet>declarations.pel-holder</Snippet></p>
-              <p><strong>{declarationAnswer(taskMeta.authority)}</strong></p>
-
-              <p><Snippet>declarations.awerb</Snippet></p>
-              <p><strong>{declarationAnswer(taskMeta.awerb)}</strong></p>
-
-              <p><Snippet>declarations.ready-for-inspector</Snippet></p>
-              <p><strong>{declarationAnswer(taskMeta.ready)}</strong></p>
-            </Fragment>
-        }
-      </StickyNavAnchor>
+      {
+        task.data.action === 'grant' && (
+          <StickyNavAnchor id="submitted-version">
+            <h2><Snippet>sticky-nav.submitted-version</Snippet></h2>
+            <p><Snippet type={task.type}>versions.submitted.hint</Snippet></p>
+            <p>
+              <Link
+                page="project.version.read"
+                className="govuk-button button-secondary"
+                versionId={submitted}
+                establishmentId={establishment.id}
+                projectId={project.id}
+                label={<Snippet>versions.submitted.label</Snippet>}
+              />
+            </p>
+            {
+              task.status === 'with-inspectorate' && (
+                <Fragment>
+                  <p><Snippet>declarations.pel-holder</Snippet></p>
+                  <p><strong>{declarationAnswer(taskMeta.authority)}</strong></p>
+
+                  <p><Snippet>declarations.awerb</Snippet></p>
+                  <p><strong>{declarationAnswer(taskMeta.awerb)}</strong></p>
+
+                  <p><Snippet>declarations.ready-for-inspector</Snippet></p>
+                  <p><strong>{declarationAnswer(taskMeta.ready)}</strong></p>
+                </Fragment>
+              )
+            }
+          </StickyNavAnchor>
+        )
+      }
 
       {
-        allDeclarationsConfirmed(task) &&
+        task.data.action === 'grant' && allDeclarationsConfirmed(task) &&
           <StickyNavAnchor id="deadline">
             <Deadline task={task} />
           </StickyNavAnchor>
       }
 
+      {
+        task.data.action === 'update' && (
+          <StickyNavAnchor id="licence-holder">
+            <h2><Snippet>sticky-nav.licence-holder</Snippet></h2>
+            <Diff values={task.data.data} model={project} schema={pick(projectSchema, 'licenceHolder')} formatters={formatters} />
+          </StickyNavAnchor>
+        )
+      }
+
+      {
+        task.data.action === 'update' && (
```

Comment:
We have the same condition twice in a row here. I'm guessing this is because the Sticky Nav needs direct children and can't handle a `<Fragment />`?

Reply:
yes. it massively sucks and I fully intend to fix it, just not right now